src/Network/Transport/TCP.hs

```diff
@@ -1450,7 +1452,7 @@ schedule theirEndPoint act = do
 -- 'runScheduledAction' or by another).
 runScheduledAction :: EndPointPair -> Action a -> IO a
 runScheduledAction (ourEndPoint, theirEndPoint) mvar = do
-    join $ readChan (remoteScheduled theirEndPoint)
+    join $ uninterruptibleMask_ $ readChan (remoteScheduled theirEndPoint)
```

Comment:
A comment proposal:
```Haskell
-- 'uninterruptibleMask_' makes `readChan` uninterruptible when the channel is not empty.
-- This is the only case we care about since 'runScheduledAction' is supposed to be called
-- after a call to 'schedule', which guarantees the channel not to be empty.
```